gits/test/capabilities_test.py

```diff
@@ -323,20 +323,54 @@ def test_cap_dch(self):
         want = s[:term._cur_x] + s[term._cur_x + 1:]
         self._check_string(want, (0, 0), (len(want), 0))
 
-    @unittest.skip('skip')
-    def test_cap_dl1(self):
-        """The terminal should have the possibility to delete a line. """
+    def test_cap_dl(self):
+        """The terminal should have the possibility to delete ``n`` number of
+        lines.
+        """
         term = self._terminal
-
-        # The y position of the cursor is on the first line.
-        self._check_cap_dl1(['s'] * term._right_most, (0, 1))
-
-        # The y position of the cursor is on the last line.
-        self._check_cap_dl1(['s'] * term._right_most, (0, term._bottom_most))
-
-        # The y position of the cursor is on an arbitrary line.
-        rand_y = random.randint(1, term._bottom_most - 1)
-        self._check_cap_dl1(['s'] * term._right_most, (0, rand_y))
+        self._check_cap_dl(0, [
+            (['f'] * term._right_most, (0, 0)),
+        ])
+
+        self._check_cap_dl(1, [
+            (['f'] * term._right_most, (0, 0)),
+        ])
+
+        self._check_cap_dl(1, [
+            (['t'] * term._right_most, (0, 0)),
+            (['a'] * term._right_most, (0, 1)),
+        ])
+
+        self._check_cap_dl(2, [
+            (['t'] * term._right_most, (0, 0)),
+            (['k'] * term._right_most, (0, 1)),
+        ])
+
+        self._check_cap_dl(2, [
+            (['f'] * term._right_most, (0, 0)),
+            (['s'] * term._right_most, (0, 1)),
+            (['k'] * term._right_most, (0, 2)),
+        ])
+
+        self._check_cap_dl(1, [
+            (['f'] * term._right_most, (0, 0)),
+            (['s'] * term._right_most, (0, 1)),
+            (['k'] * term._right_most, (0, 2)),
+        ])
+
+        self._check_cap_dl(0, [
+            (['f'] * term._right_most, (0, 0)),
+            (['s'] * term._right_most, (0, 1)),
+            (['k'] * term._right_most, (0, 2)),
+        ])
+
+        lines_cnt = random.randint(2, term._bottom_most - 1)
```

Comment:
`lines_cnt` is a random number of lines to be put on the screen, so let's name it `lines_number`.